server/src/index.ts

```diff
@@ -119,6 +128,15 @@ const createTransport = async (req: express.Request): Promise<Transport> => {
       const value = req.headers[key];
       headers[key] = Array.isArray(value) ? value[value.length - 1] : value;
     }
+    
+    // If the header "x-custom-auth-header" is present, use its value as the custom header name.
+    if (req.headers["x-custom-auth-header"] !== undefined) {
+      const customHeaderName = req.headers["x-custom-auth-header"] as string;
+      if (req.headers[customHeaderName.toLowerCase()] !== undefined) {
+        const value = req.headers[customHeaderName.toLowerCase()];
```

Comment:
Avoid calling lowerCase twice.
```suggestion
      const lcHeaderName = customHeaderName.toLowerCase();
      if (req.headers[lcHeaderName] !== undefined) {
        const value = req.headers[lcHeaderName];
```

Reply:
resolved